Sming/system/m_printf.cpp

```diff
@@ -11,39 +11,33 @@ Descr: embedded very simple version of printf with float support
 
 #define MPRINTF_BUF_SIZE 256
 
-#define OVERFLOW_GUARD 24
-
 static void defaultPrintChar(uart_t *uart, char c) {
-	return uart_tx_one_char(c);
+    return uart_tx_one_char(c);
```

Comment:
> That's why I converted all tabs to space before committing.

Changing of formatting and coding style can happen in another PR. This PR is only related to `m_vsnprintf` changes.